benchmarks/recursive_computation/scenario.py

```diff
@@ -0,0 +1,70 @@
+import time
+from typing import Callable
+from typing import Generator
+
+import bm
+import bm.utils as utils
+
+from ddtrace.trace import tracer
+
+
+class RecursiveComputation(bm.Scenario):
+    name: str
+    max_depth: int
+    enable_sleep: bool
+    sleep_duration: float
+    profiler_enabled: bool
+
+    def cpu_intensive_computation(self, depth: int) -> int:
+        limit = 1000 + (depth * 100)
+        primes = []
+
+        for num in range(2, limit):
+            is_prime = True
+            for i in range(2, int(num**0.5) + 1):
+                if num % i == 0:
+                    is_prime = False
+                    break
+
+            if is_prime:
+                primes.append(num)
+
+        return len(primes)
+
+    def recursive_traced_computation(self, depth: int = 0) -> int:
+        with tracer.trace(f"recursive_computation.depth_{depth}") as span:
+            span.set_tag("recursion.depth", depth)
+            span.set_tag("recursion.max_depth", self.max_depth)
+            span.set_tag("profiler.enabled", self.profiler_enabled)
+            span.set_tag("component", "recursive_computation")
+
+            start_time = time.time()
+            result = self.cpu_intensive_computation(depth)
```

Comment:
you might also want to reduce the number of times we call this cpu_intensive_computation.
It's still taking more than 10 mins
https://gitlab.ddbuild.io/DataDog/apm-reliability/dd-trace-py/-/jobs/975894154